ocsp.go

```diff
@@ -512,18 +512,18 @@ func isValidOCSPStatus(status ocspStatusCode) bool {
 // verifyPeerCertificate verifies all of certificate revocation status
 func verifyPeerCertificate(verifiedChains [][]*x509.Certificate) (err error) {
 	for i := 0; i < len(verifiedChains); i++ {
-		n := len(verifiedChains[i]) - 1
-		if !verifiedChains[i][n].IsCA || string(verifiedChains[i][n].RawIssuer) != string(verifiedChains[i][n].RawSubject) {
+		numberOfNoneRootCA := len(verifiedChains[i]) - 1
+		if !verifiedChains[i][numberOfNoneRootCA].IsCA || string(verifiedChains[i][numberOfNoneRootCA].RawIssuer) != string(verifiedChains[i][numberOfNoneRootCA].RawSubject) {
```

Comment:
Add comment - "Check if the last Non Root Cert is also a CA or is self signed." Also I am not sure if we need to perform the second check as the boolean IsCA is set if the certificate is self signed.